timelock-impl/src/test/java/com/palantir/atlasdb/timelock/batch/MultiClientConjureTimelockResourceTest.java

```diff
@@ -91,6 +105,41 @@ public void requestThrowsIfAnyQueryFails() {
                 .isInstanceOf(BlockingTimeoutException.class);
     }
 
+    @Test
+    public void canStartTransactionsForMultipleClients() {
+        List<String> namespaces = ImmutableList.of("client1", "client2");
+        assertThat(Futures.getUnchecked(
+                        resource.startTransactions(AUTH_HEADER, getStartTransactionsRequests(namespaces))))
+                .isEqualTo(getStartTransactionsResponseList(namespaces));
+    }
+
+    private Map<Namespace, ConjureStartTransactionsResponse> getStartTransactionsResponseList(List<String> namespaces) {
+        return KeyedStream.of(namespaces)
+                .map(namespace -> mockedStartTxnResponse())
+                .mapKeys(Namespace::of)
+                .collectToMap();
+    }
+
+    private Map<Namespace, ConjureStartTransactionsRequest> getStartTransactionsRequests(List<String> namespaces) {
+        return KeyedStream.of(namespaces)
+                .map(namespace -> ConjureStartTransactionsRequest.builder()
+                        .numTransactions(5)
+                        .requestId(UUID.randomUUID())
+                        .requestorId(UUID.randomUUID())
+                        .build())
+                .mapKeys(Namespace::of)
+                .collectToMap();
+    }
+
+    private ConjureStartTransactionsResponse mockedStartTxnResponse() {
```

Comment:
Hmm, I get that all the fields here are mocked, but the object itself is not. Maybe worth a rename? Just `fakeStartTxnResponse` or even just `startTransactionResponse`? I don't know this one, but does this need to be a method, or does it work if you make it a field? Maybe the mocking doesn't work that way, not 100% sure there.